codegen/masm/src/config.rs

```diff
@@ -0,0 +1,35 @@
+pub struct CodegenConfig {
+    // Memory location of the trace length using the following format:
+    //
+    //      [trace_len_address] => [0, 0, 0, trace_len]
```

Comment:
I believe in memory order it would be `[trace_len, 0, 0, 0]`, but in stack-order it is as documented. AFAIK the memory order is not observable by the code, so this documents what the code should expect, in particular, if using `mem_loadw` the result is as above.

Reply:
I think for documentation purposes we should not use stack order. For example, when we refer to `mem[0][0]` we refer to the first element at memory address 0. This would be the element loaded by `mem_load` instruction. Listing it last makes things a bit more confusing.

I think it is OK to list memory values in the "natural" order because we know that semantics of `mem_loadw` instruction is to put elements onto the stack in the "stack order".

Reply:
The way I think about this is in terms of least significant vs. most significant elements. In a word, we usually say that the least significant element is `word[0]` and the most significant is `word[3]`. On the stack, we say that the least significant element is in the deeper in the stack than than the most significant.

We use the same with non-word values too. For example, when representing a `u64` value using two elements, the element with least significant bits is deeper in the stack than the element with the most significant bits. And when we store it in memory the list significant element ends up in `word[0]`.

I'd prefer to keep this way of looking at things as it would be consistent with how we do it in other places.

Reply:
updated